blockchain/chain/src/store/chain_store.rs

```diff
@@ -57,18 +61,27 @@ where
             .map(Arc::new);
         Self {
             db,
+            publisher: Publisher::new(SINK_CAP),
             tip_index: TipIndex::new(),
             heaviest,
         }
     }
 
     /// Sets heaviest tipset within ChainStore and store its tipset cids under HEAD_KEY
-    pub fn set_heaviest_tipset(&mut self, ts: Arc<Tipset>) -> Result<(), Error> {
+    pub async fn set_heaviest_tipset(&mut self, ts: Arc<Tipset>) -> Result<(), Error> {
         self.db.write(HEAD_KEY, ts.key().marshal_cbor()?)?;
         self.heaviest = Some(ts);
+        self.publisher
+            .publish(self.heaviest.as_ref().unwrap().clone())
+            .await;
         Ok(())
     }
 
+    // subscribing returns a future sink that we can essentially iteratre over using future streams
```

Comment:
```suggestion
    // subscribing returns a future sink that we can essentially iterate over using future streams
```
Is iterate the word you meant to put here?